File: backend/app/schemas/document.py
import uuid
from datetime import date, datetime

from pydantic import ConfigDict

from app.models.enums import DocumentStatus
from app.schemas.base import AppBaseModel


class DocumentRead(AppBaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    company_id: uuid.UUID
    filename: str
    file_type: str
    status: DocumentStatus
    period_start: date | None
    period_end: date | None
    error_message: str | None
    source_type: str
    created_at: datetime

File: frontend/src/types/document.ts
export type DocumentStatus = "pending" | "processing" | "extracted" | "failed";

export const IN_PROGRESS_DOCUMENT_STATUSES: DocumentStatus[] = ["pending", "processing"];

export type DocumentSourceType = "manual_upload" | "auto_fetched";

export interface CompanyDocument {
  id: string;
  company_id: string;
  filename: string;
  file_type: string;
  status: DocumentStatus;
  period_start: string | null;
  period_end: string | null;
  error_message: string | null;
  source_type: DocumentSourceType;
  created_at: string;
}
